src/client/Components/Util/OAuth.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import PropTypes from 'prop-types';
+
+export default class OAuth extends React.Component {
+  state = {
+    user: {},
+    disabled: ''
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.startAuth = this.startAuth.bind(this);
+    this.closeCard = this.closeCard.bind(this);
+  }
+
+  componentDidMount() {
+    const { socket, provider } = this.props;
```

Comment:
Where do these come from?

Reply:
If you check the Sign-In page, you'll see two properties in the setting for the OAuth component. That's how React handles passing values into components.